org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/GuardedPattern.java

```diff
@@ -199,6 +199,16 @@ public static List propertyDescriptors(int apiLevel, boolean previewEnabled) {
 	 */
 	public Expression getExpression() {
 		supportedOnlyIn21();
+		if(this.conditonalExpression == null) {
+            //lazy init must be thread-safe for readers
+            synchronized (this) {
+                if( this.conditonalExpression == null) {
+                    preLazyInit();
+                    this.conditonalExpression = this.ast.newInfixExpression();
```

Comment:
Please use null literal
                    this.conditonalExpression = this.ast.newNullLiteral();